---
title: "Publication-ready Tables with R"
subtitle: "Make Your Research Easy‚ÄîStop Wasting Time on Formatting. Learn how to create, customize, and export stunning publication-ready tables for your research in just a few lines of code!"
editor: source
---

-------------------------------------------------------------------
üìä An all-in-one, interactive workshop designed to help you create stunning, publication-ready tables effortlessly with gtsummary.

üìÖ Friday \& March, February 5 \&8, 2025 | 9:00pm - 12:00 00pm

üíª Mode of Instructions (Virtual/Zoom)

‚úçÔ∏è **Register** with [Google Form](https://forms.gle/a7dDYNbraBDFFAw19)

üí∞ **Registration Fee**: 310 tk

üìß After registration, you will receive a confirmation email within 24 hours with further instructions to join the workshop.

------------------------------------------------------------------------

## Workshop Overview
In this workshop, you‚Äôll learn how to effortlessly create publication-ready tables for your research papers using the powerful `{gtsummary}` package in R. Designed for researchers. This workshop will guide you through the process of transforming raw data into clean, well-organized, and visually appealing tables‚Äîwithout spending excessive time on formatting.

Through interactive demonstrations and hands-on exercises, you will gain a solid understanding of how to summarize research data efficiently, generate insightful reports, and ensure your results are presentation-ready for scientific journals. By the end of the workshop, you‚Äôll be equipped with the skills and tools needed to streamline your data reporting process and focus on what matters most‚Äîyour research.

## Let's make great tables!
This online workshop will only take a few hours of your time but will drastically improve your skills in creating publication-ready tables with `{gtsummary}` in R.

By the end, you‚Äôll become the go-to expert in your team when it comes to summarizing and presenting data in a way that‚Äôs both clear and visually appealing. You'll generate tables that are not only easy to interpret but also publication-ready, eliminating the hassle of endless formatting.

Sure, you could spend hours piecing together tips and tricks from blogs, forums, and documentation, but we've condensed all the essential techniques into one streamlined workshop to save you time and effort.

This ultimate workshop on publication-ready tables with gtsummary is designed to make your research reporting process easier and more efficient than ever.

![](images/gtsummary.gif)

## Master Publication-Ready Tables with gtsummary, step by step
You will begin with basic table generation and progress to creating the most customized and visually appealing clinical reports with gtsummary in R.

Each module also includes detailed explanations of key gtsummary functions and essential data summarization concepts, ensuring that by the end, you'll be equipped to create professional-quality tables for any research paper.

Here is the plan:

::: {.grid}

::: {.g-col-12 .g-col-md-6}
<div class="card h-100" id="card-first-color">
<div class="card-body d-flex flex-column">
<p class="card-text">Module 1</p>
<p class="card-text"><b>Introduction & Setup</p></b>
<p class="card-text">Learn why R is the best tool for automating table creation and ensuring publication-quality formatting using **gtsummary** and **gt**. Set up R, install required packages, and prepare datasets for table generation.</p>

</div>
</div>
:::

::: {.g-col-12 .g-col-md-6}
<div class="card h-100" id="card-first-color">
<div class="card-body d-flex flex-column">
<p class="card-text">Module 2</p>
<p class="card-text"><b>Descriptive Tables</p></b>
<p class="card-text">Create well-structured summary tables for **categorical** and **continuous** variables. The module will also cover how to create **cross-tabulation (contingency)** tables to examine relationships between categorical variables.</p>

</div>
</div>
:::

::: {.g-col-12 .g-col-md-6}
<div class="card h-100" id="card-first-color">
<div class="card-body d-flex flex-column">
<p class="card-text">Module 3</p>
<p class="card-text"><b>Difference Tables</p></b>
<p class="card-text">In this module, you'll learn how to create Difference Tables that summarize statistical comparisons, such as **t-tests** and **ANOVA** results, between different groups. These tables are essential for presenting the differences in means or other summary statistics across categorical variables.</p>

</div>
</div>
:::

::: {.g-col-12 .g-col-md-6}
<div class="card h-100" id="card-first-color">
<div class="card-body d-flex flex-column">
<p class="card-text">Module 4</p>
<p class="card-text"><b>Association Tables</p></b>
<p class="card-text">In this module, you will learn how to create Association Tables that summarize the relationship between categorical variables, such as those tested using **chi-square** tests. These tables are essential for presenting statistical associations between different groups or factors.</p>

</div>
</div>
:::

::: {.g-col-12 .g-col-md-6}
<div class="card h-100" id="card-first-color">
<div class="card-body d-flex flex-column">
<p class="card-text">Module 5</p>
<p class="card-text"><b>Regression Analysis Tables</p></b>
<p class="card-text">In this module, you'll learn how to create Regression Analysis Tables for various types of regression models, including **Linear Regression**, **Logistic Regression**, **Ordinal Regression**, and **Poisson Regression**. These tables are key for summarizing model results, providing a clear overview of the relationships between predictors and outcomes.</p>

</div>
</div>
:::

::: {.g-col-12 .g-col-md-6}
<div class="card h-100" id="card-first-color">
<div class="card-body d-flex flex-column">
<p class="card-text">Module 6</p>
<p class="card-text"><b>Survival Analysis Tables</p></b>
<p class="card-text">In this module, you will learn how to create Survival Analysis Tables that summarize the results of time-to-event analyses, such as **Cox Proportional Hazards** and **Kaplan-Meier survival curves**. These tables are crucial for presenting **survival rates**, **hazard ratios**, and associated statistics in clinical and epidemiological research.</p>

</div>
</div>
:::
:::

## Learn how to make this!
### Descriptive Tables 
Descriptive statistics for continuous, categorical, and dichotomous variables in R, and presents the results in a beautiful, customizable summary table ready for publication (for example, Table 1 or demographic tables).


```{r}
#| label: load-data-pkg 
#| echo: false
#| warning: false
#| message: false
# load data 
library(tidyverse)
library(gtsummary)
library(gt)
data <- readxl::read_excel("./code/data/KAP_Thals_Processed.xlsx", sheet = 5)
```

```{r}
#| label: descriptive-tables 
#| echo: false
# descriptive tables 
data %>% 
  select(2:7) %>% 
  tbl_summary(statistic = all_continuous() ~ "{mean} ({sd})", 
              type = all_dichotomous() ~ "categorical") |> 
  bold_labels() |> 
  as_kable_extra()
```

### Comparative Tables 
Comparative tables are a type of analytical table used to present and compare data across different variables, categories, or entities. These tables are commonly employed to highlight similarities, differences, and relationships between data points.They allow for a concise and structured way to showcase information side by side, making it easier for the audience to draw insights and conclusions.  

```{r}
#| label: comparative tables
#| echo: false
# descriptive tables 
data <- data %>% filter(`Do you know about thalassemia?` == "Yes")  

data %>% 
  select(2:6, KnowledgeLevelBinary) %>% 
  tbl_summary(statistic = all_continuous() ~ "{mean} ({sd})", 
              type = all_dichotomous() ~ "categorical", 
              by = KnowledgeLevelBinary) %>% 
  add_overall() %>% 
  bold_labels() |> 
  as_kable_extra()
```



### Association Tables
Analytical tables typically refer to organized and structured data representations used for analysis, comparison, and interpretation of information. They are commonly used in various fields such as statistics, research, business, and academia. Analytical tables help present data in a clear and concise manner, making it easier to identify patterns, trends, and relationships among variables.


```{r}
#| label: analytical-tables 
#| echo: false
# descriptive tables 
data |> 
  select(2:6, KnowledgeLevelBinary) |> 
  tbl_summary(statistic = all_continuous() ~ "{mean} ({sd})", 
              type = all_dichotomous() ~ "categorical", 
              by = KnowledgeLevelBinary) |> 
  add_overall() |> 
  add_p() |> 
  bold_p(t = 0.05) |> 
  bold_labels() |> 
  as_kable_extra()
```

### Univariate Regression Tables 
Univariate regression is a type of linear regression where you have one independent variable (predictor) and one dependent variable (response). Regression tables are used to present the results of regression analysis, including coefficients, standard errors, significance levels, and other relevant statistics. Below is an example of what a simple univariate regression table might look like:
```{r}
#| label: uv-regression
#| echo: false
#| message: false
# descriptive tables 
trial |> 
  select(response, age, trt, marker, stage, grade) |> 
  tbl_uvregression(
    method = glm,
    y = response,
    method.args = list(family = binomial),
    exponentiate = TRUE,
    pvalue_fun = ~ style_pvalue(.x, digits = 2)
  ) |> 
  add_global_p() |> 
  add_nevent() |> 
  add_q() |> 
  bold_p() |> 
  bold_p(t = 0.10, q = TRUE) |> 
  bold_labels() |> 
  as_kable_extra()
```

### Multivariate Regression Tables 
A multivariate regression table provides a summary of the results obtained from a multivariate regression analysis. This type of analysis involves multiple independent variables (predictors) and a single dependent variable. 

```{r}
#| label: mv-regression
#| echo: false
#| message: false
# descriptive tables 
m1 <- glm(response ~ age + stage + trt + marker + grade, trial, family = binomial)
m1 |> 
  tbl_regression(
    exponentiate = TRUE,
    pvalue_fun = ~ style_pvalue(.x, digits = 2),
  ) |> 
  add_global_p() |> 
  bold_p(t = 0.05) |> 
  bold_labels() |> 
  italicize_levels() |> 
  as_kable_extra()
```

## Prework

Before attending the workshop please have the following installed and configured on your machine.

-   Recent version of **R** 

-   Recent version of **RStudio**

-   Most recent release of the [**gtsummary**](https://www.danieldsjoberg.com/gtsummary/) and other packages used in workshop.

    ```r
    instll_pkgs <- 
      c("gtsummary", "tidyverse", "labelled", "usethis", 
        "causaldata", "fs", "skimr", "car", "emmeans")
    install.packages(instll_pkgs)
    ```
-   Ensure you can knit R markdown documents
    - Open RStudio and create a new Rmarkdown document
    - Save the document and check you are able to knit it.

 

## Instructor 
### Bio
![](images/JHossain.jpg){style="float:right;padding: 0 0 0 10px;" fig-alt="Headshot of Jubayer" width="200"} 

I am a biomedical researcher focusing on neuroepidemiology, cancer epidemiology, and cancer bioinformatics. My primary interest lies in applying machine learning algorithms to health data including electronic health records, multi-omics big data, and neurological images. I aim to optimize clinically actionable biomarkers and develop predictive models to enable early diagnosis and treatment for neurological disorders and cancer. Furthermore, I am interested in visualizing these data and models to facilitate biological interpretation and clinical applications.

I am currently the Founder and Executive Director of [CHIRAL Bangladesh](https://chiralbd.org/). My leadership extends to [aiHealth Lab](https://aihealthlab.owlstown.net/) and Big Bioinformatics Lab, where the focus is on advancing cancer bioinformatics and artificial intelligence for health. As an educator, I hold the position of Lead Instructor and Organizer at the Training Unit of CHIRAL. Additionally, I serve as an instructor for the Data Science for Biologists program at cBLAST.

To further bridge data science with biomedical research, I have initiated [DataViz Gallery](https://hossainlab.github.io/dataviz-gallery/), a curated collection of compelling data visualizations using Python and R. This initiative aims to enhance the interpretability of complex biological and clinical data through effective visual storytelling. 

This [page](https://hossainlab.owlstown.net/) highlights my teaching and research projects. Please reach out if you want to collaborate or have questions.

### Skills 
**Programming Languages:** Python, R, SQL, Julia, JavaScript; **Data Science:** scikit-learn, PyCaret, Dask, PySpark; **GIS & Remote Sensing:** ArcGIS, Geopandas, Xarray, Giovani; **Analytics Softwares:** SPSS, PowerBI, Microsoft Excel; **Survey Tools:** RedCap, KoboToolBox, EpiCollect, Google Forms; **Academic Writing Tools:** Microsoft Word, LaTeX, Mendeley; **Bioinformatics:** BioPython, Bioconductor, BioPandas, Galaxy, NGS, RNASeq, ssRNASeq; **Miscellaneous Skills:** UNIX, Version Control(Git), Web Scraping, APIs.


### Selected Publications 
1. **Hossain, M.J.**, Das, M. & Munni, U.R.  
   *Urgent call for compulsory premarital screening: a crucial step towards thalassemia prevention in Bangladesh.*  
   Orphanet J Rare Dis 19, 326 (2024). [DOI: 10.1186/s13023-024-03344-1](https://doi.org/10.1186/s13023-024-03344-1)

2. Shanta, A.S.; Islam, N.; Al Asad, M.; Akter, K.; Habib, M.B.; **Hossain, M.J.**; Nahar, S.; Godman, B.; Islam, S.  
   *Resistance and Co-Resistance of Metallo-Beta-Lactamase Genes in Diarrheal and Urinary-Tract Pathogens in Bangladesh.*  
   Microorganisms 2024, 12, 1589. [DOI: 10.3390/microorganisms12081589](https://doi.org/10.3390/microorganisms12081589)

3. Islam, M.W., Shahjahan, M., Azad, A.K., **Hossain, M.J.**  
   *Factors contributing to antibiotic misuse among parents of school-going children in Dhaka City, Bangladesh.*  
   Sci Rep 14, 2318 (2024). [DOI: 10.1038/s41598-024-52313-y](https://doi.org/10.1038/s41598-024-52313-y)

4. **Hossain, M.J.**, Islam, M.W., Munni, U.R., Gulshan, R., Mukta, S.A., Miah, M.S., Sultana, S., Karmakar, M., Ferdous, J., & Islam, M.A.  
   *Health-related quality of life among thalassemia patients in Bangladesh using the SF‚Äë36 questionnaire.*  
   Scientific Reports, 13(1). [DOI: 10.1038/s41598-023-34205-9](https://doi.org/10.1038/s41598-023-34205-9)

5. Towhid, S.T., **Hossain, M.J.**, Sammo, M.A.S., & Akter, S.  
   *Perception of Students on Antibiotic Resistance and Prevention: An Online, Community-Based Case Study from Dhaka, Bangladesh.*  
   European Journal of Biology and Biotechnology, 3(3), 14‚Äì19. [DOI: 10.24018/ejbio.2022.3.3.341](https://doi.org/10.24018/ejbio.2022.3.3.341)

6. **Hossain, M.J.**, Towhid, S.T., Sultana, S., Mukta, S.A., Gulshan, R., Miah, M.S.  
   *Knowledge and Attitudes towards Thalassemia among Public University Students in Bangladesh.*  
   Microbial Bioactives, 5(2). [DOI: 10.25163/microbbioacts.526325](https://doi.org/10.25163/microbbioacts.526325)

7. **Hossain, M.J.**, Towhid, S.T., Akter, S., Shahjahan, M., Roy, T., Akter, B., & Nodee, T.A.  
   *Knowledge and Self-Management Practice Among Diabetic Patients from the Urban Areas in Bangladesh.*  
   Journal of Angiotherapy, 7(1), 1-10. [DOI: 10.25163/angiotherapy.717340](https://doi.org/10.25163/angiotherapy.717340)

8. **Hossain, M.J.**, Das, M., Islam, M.W., Shahjahan, M., Ferdous, J.  
   *Community engagement and social participation in dengue prevention: A cross-sectional study in Dhaka City.*  
   Health Sci Rep. 2024; 7:e2022. [DOI: 10.1002/hsr2.2022](https://doi.org/10.1002/hsr2.2022)

9. **Hossain, M.J.**, Azad, A.K., Shahid, M.S.B., Shahjahan, M., Ferdous, J.  
   *Prevalence, antibiotic resistance pattern for bacteriuria from patients with urinary tract infections.*  
   Health Sci Rep. 2024; 7:e2039. [DOI: 10.1002/hsr2.2039](https://doi.org/10.1002/hsr2.2039)